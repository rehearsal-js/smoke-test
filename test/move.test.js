import { describe, expect, test } from 'vitest';
import { getProjectFixture, setupProjectRunner } from './test-helpers';

describe('validation-test: rehearsal move', async () => {
  const variants = [
    { variant: 'simple', args: ['--help'] },
    //   { variant: 'simple', args: [] },
    { variant: 'simple', args: ['--source', 'index.js'] },
    { variant: 'workspace', args: ['--childPackage', 'packages/blorp'] }
  ];

  test.each(variants)('$variant: move $args', async ({ variant, args }) => {
    const project = await getProjectFixture(variant);
    const { run } = setupProjectRunner(project);
    const results = run(['move', ...args]);
    expect(results.exitCode).toBe(0);
    expect(results.stdout).toMatchSnapshot();
    project.dispose();
  });
});
