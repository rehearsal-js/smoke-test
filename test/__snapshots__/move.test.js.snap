// Vitest Snapshot v1

exports[`validation-test: rehearsal move > 'simple': move [ '--help' ] 1`] = `
"Usage: rehearsal move|mv [options]

git mv conversion of JS files -> TS files

Options:
  -s, --source <path to source file | directory>  the explicit source file OR explicit directory with implicit child directories moved. migration strategy is ignored (default: \\"\\")
  -p, --childPackage <path to child package>      specify the child-package relative to process.cwd(). migration strategy is leveraged moving all necessary files in the dependency graph for this package (default: \\"\\")
  -d, --dryRun                                    Do nothing; only show what would happen (default: false)
  -h, --help                                      display help for command"
`;

exports[`validation-test: rehearsal move > 'simple': move [ '--source', 'index.js' ] 1`] = `
"[32minfo[39m:    @rehearsal/move VERSION
[STARTED] Initialize
[SUCCESS] Initialize
[STARTED] Executing git mv
[DATA] git mv failed, using mv
[DATA] renamed: 
[DATA] /index.js -> /index.ts
[DATA] 
[SUCCESS] Executing git mv"
`;

exports[`validation-test: rehearsal move > 'workspace': move [ '--childPackage', 'packages/blorp' ] 1`] = `
"[32minfo[39m:    @rehearsal/move VERSION
[STARTED] Initialize
[SUCCESS] Initialize
[STARTED] Analyzing project dependency graph
[SUCCESS] Analyzing project dependency graph
[STARTED] Executing git mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] git mv failed, using mv
[DATA] renamed: 
[DATA] /packages/blorp/build.js -> /packages/blorp/build.ts
[DATA] /packages/blorp/lib/impl.js -> /packages/blorp/lib/impl.ts
[DATA] /packages/blorp/index.js -> /packages/blorp/index.ts
[DATA] 
[SUCCESS] Executing git mv"
`;
